soc/arm/atmel_sam/sam4e/soc.h

```diff
@@ -38,6 +40,9 @@
 #include "../common/soc_pmc.h"
 #include "../common/soc_gpio.h"
 
+/* Add include for DTS generated information */
+#include <devicetree.h>
```

Comment:
I suppose you are including `devicetree.h` here (inside `#ifndef _ASMLANGUAGE` because of the DT symbol references below.
```
/** Processor Clock (HCLK) Frequency */
#define SOC_ATMEL_SAM_HCLK_FREQ_HZ DT_ARM_CORTEX_M4F_0_CLOCK_FREQUENCY
/** Master Clock (MCK) Frequency */
#define SOC_ATMEL_SAM_MCK_FREQ_HZ SOC_ATMEL_SAM_HCLK_FREQ_HZ
```

In that case, one of the following should be done:
1. Move the defs into `#ifndef _ASMLANGUAGE` if these defs need not be referenced in ASM code OR
2. Move the inc to outside `#ifndef _ASMLANGUAGE` so that these defs are also valid in ASM code.

Reply:
I'm follow same approach found on SAME70/SAMV71 but is clear that I need fix that too :D